ee/tabby-ui/app/(dashboard)/activities/components/activity.tsx

```diff
@@ -0,0 +1,333 @@
+                ]}
+                defaultValue={DEFAULT_DATE_RANGE}
+                onSelect={updateDateRange}
+                hasToday
+                hasYesterday
+              />
+            </div>
+
+            <Card x-chunk="dashboard-06-chunk-0" className="bg-transparent">
+              {(!data?.userEvents.edges ||
+                data?.userEvents.edges.length === 0) && (
+                <CardContent className="flex flex-col items-center py-40 text-sm">
+                  <IconFileSearch className="mb-2 h-10 w-10" />
+                  <p className="font-semibold">
+                    No data available for the chosen dates
+                  </p>
+                  <p className="text-muted-foreground">
+                    Please try a different date range
+                  </p>
+                </CardContent>
+              )}
+
+              {data?.userEvents.edges && data?.userEvents.edges.length > 0 && (
+                <>
+                  <CardContent className="w-[calc(100vw-4rem)] overflow-x-auto pb-0 md:w-auto">
+                    <Table>
+                      <TableHeader>
+                        <TableRow>
+                          <TableHead className="md:w-[25%]">Event</TableHead>
+                          <TableHead className="md:w-[25%]">People</TableHead>
+                          <TableHead className="md:w-[25%]">Time</TableHead>
+                          <TableHead className="md:w-[25%]">Language</TableHead>
+                        </TableRow>
+                      </TableHeader>
+                      <TableBody>
+                        {userEvents?.edges
+                          .sort(
+                            (a, b) =>
+                              new Date(b.node.createdAt).getTime() -
+                              new Date(a.node.createdAt).getTime()
+                          )
+                          .map(userEvent => (
+                            <ActivityRow
+                              key={userEvent.cursor}
+                              activity={userEvent.node}
+                            />
+                          ))}
+                      </TableBody>
+                    </Table>
+                  </CardContent>
+                </>
+              )}
+            </Card>
+
+            {(data?.userEvents.pageInfo?.hasNextPage ||
+              data?.userEvents.pageInfo?.hasPreviousPage) && (
+              <div className="flex justify-end">
+                <div className="flex w-[100px] items-center justify-center text-sm font-medium">
+                  Page {page}
+                </div>
+                <div className="flex items-center space-x-2">
+                  <Button
+                    variant="outline"
+                    className="h-8 w-8 p-0"
+                    disabled={!data?.userEvents.pageInfo?.hasNextPage}
+                    onClick={e => {
+                      setQueryVariables({
```

Comment:
I recommend using relayPagination(), let's discuss offline

Reply:
Checked with Jueliang, relayPagination seems not suitable for this case